Source/WebCore/style/PropertyAllowlist.cpp

```diff
@@ -118,6 +122,317 @@ bool isValidMarkerStyleProperty(CSSPropertyID id)
     return false;
 }
 
+bool isValidFirstLetterStyleProperty(CSSPropertyID id)
+{
+    switch (id) {
+    case CSSPropertyBackground:
```

Comment:
Not a necessary change, but I would love for these to be encoded in CSSProperties.json instead by adding two new entries for the "codegen-properties" dictionary: `"valid-for-first-letter-psuedo-elements": true` and `"valid-for-first-line-psuedo-elements": true` (though feel free to change the name to something more concise and/or swap it to the negative, e.g. "invalid-for-first-letter-psuedo-elements").